### PR TITLE
Improve IconButton

### PR DESCRIPTION
**Pull request by sliptype**
### Description

This PR:
  * Moves `IconButton` to `core`
  * Adds a `fill` prop to `IconButton` to allow for custom icon colors
  * Updates the `IconButton` style props
  * Fixes the DynamicImage in the track screen header (broken as part of the merge, and missed)

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

Tested on ios simulator

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### CODE REVIEW

File: src/components/core/IconButton.tsx [hunk 1]
```diff
@@ -3,46 +3,53 @@ import { TouchableOpacity } from 'react-native-gesture-handler'
...
       fillSecondary?: string
     }
   >
-  containerStyle?: StyleProp<ViewStyle>
-  style?: StyleProp<ViewStyle>
-  isActive?: boolean
   isDisabled?: boolean
+  onPress?: () => void
+  styles?: {
+    root?: StyleProp<ViewStyle>
```
Comment (dylanjeffers) [inline src/components/core/IconButton.tsx:19]: just a headsup, got a slight helper for this in /app/styles/types, down to improve on it to also include the `style?:` addition, so we get both

File: src/components/track-tile/TrackTileActionButtons.tsx [hunk 2]
```diff
@@ -165,35 +164,21 @@ export const TrackTileActionButtons = ({
...
-        />
-      )}
+      styles={{ root: styles.button }}
     />
   )
 
   const moreButton = (
     <IconButton
-      onPress={onPressOverflow}
+      fill={neutralLight4}
```
Comment (dylanjeffers) [inline src/components/track-tile/TrackTileActionButtons.tsx:177]: this is all so much cleaner!

File: src/screens/track-screen/TrackScreenHeader.tsx [hunk 3]
```diff
@@ -421,14 +421,14 @@ export const TrackScreenHeader = ({
       // userId={coSign.user.user_id}
     >
       <DynamicImage
-        image={{ uri: image }}
-        style={[styles.coverArt] as ImageStyle[]}
+        source={{ uri: image }}
```
Comment (dylanjeffers) [inline src/screens/track-screen/TrackScreenHeader.tsx:424]: thanks for the fix, weird didnt catch this with typecheck earlier...

**Review by dylanjeffers — APPROVED** [review]
these changes look great!

File: src/screens/track-screen/TrackScreenHeader.tsx (hunk 3 above)
Comment (sliptype) [reply src/screens/track-screen/TrackScreenHeader.tsx:424]: Idk why it didn't catch, but it was my fault bc I merged track screen after the dynamic image change

File: src/components/core/IconButton.tsx (hunk 1 above)
Comment (sliptype) [reply src/components/core/IconButton.tsx:19]: Dope, I can add that in another pr

File: src/screens/track-screen/TrackScreenHeader.tsx (hunk 3 above)
Comment (dylanjeffers) [reply src/screens/track-screen/TrackScreenHeader.tsx:424]: ah, makes sense

**Pull request merged**